diff --git a/package.json b/package.json
index 46145c7..bd1047f 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "svg-sandbox",
   "version": "1.0.0",
-  "main": "index.js",
+  "main": "index.jsx",
   "license": "MIT",
   "scripts": {
     "dev": "webpack-dev-server"
diff --git a/src/components/App.jsx b/src/components/App.jsx
new file mode 100644
index 0000000..e583685
--- /dev/null
+++ b/src/components/App.jsx
@@ -0,0 +1,53 @@
+import React from 'react';
+import AugmentedVideoPlayer from '../augmented-video-player/components/augmented-video-editor';
+
+class App extends React.Component {
+  componentDidMount() {
+    const app = this.div;
+
+    const player = new AugmentedVideoPlayer();
+    player.videoSource = 'http://localhost:3000/video';
+
+    app.appendChild(player.el);
+
+    const button = document.createElement('button');
+    button.innerHTML = 'play';
+
+    button.addEventListener('click', () => {
+      player.play();
+    });
+
+    app.appendChild(button);
+
+    const pauseButton = document.createElement('button');
+    pauseButton.innerHTML = 'pause';
+
+    pauseButton.addEventListener('click', () => {
+      player.pause();
+    });
+
+    app.appendChild(pauseButton);
+
+    const addTagButton = document.createElement('button');
+    addTagButton.innerHTML = 'add tag';
+
+    addTagButton.addEventListener('click', () => {
+      player.addTag();
+    });
+
+    app.appendChild(addTagButton);
+  }
+
+  render() {
+    return (
+      <div
+        className="app"
+        ref={div => {
+          this.div = div;
+        }}
+      />
+    );
+  }
+}
+
+export default App;
diff --git a/src/draggableness/draggableness.js b/src/draggableness/draggableness.js
index 83797ef..c6724e9 100644
--- a/src/draggableness/draggableness.js
+++ b/src/draggableness/draggableness.js
@@ -7,6 +7,7 @@ class Draggableness {
     this.dragCallback = dragCallback;
     this.releaseCallback = releaseCallback;
     [this.draggable] = Draggable.create(this.target, {
+      dragClickables: true,
       bounds,
       onPress: this.onPress,
       onRelease: this.onRelease,
diff --git a/src/index.jsx b/src/index.jsx
new file mode 100644
index 0000000..c6ce28e
--- /dev/null
+++ b/src/index.jsx
@@ -0,0 +1,19 @@
+import React from 'react';
+import { render } from 'react-dom';
+import './style.css';
+import App from './components/App';
+import '../reset.css';
+import '../node_modules/regenerator-runtime/runtime';
+
+const renderApp = () => {
+  const app = document && document.getElementById('app');
+  if (app) render(<App />, app);
+};
+
+renderApp();
+
+if (module.hot) {
+  module.hot.accept('./components/App', () => {
+    renderApp();
+  });
+}
diff --git a/webpack.config.js b/webpack.config.js
index 89ab610..cde10cf 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -11,7 +11,7 @@ module.exports = {
     'react-hot-loader/patch',
     'webpack-dev-server/client?http://localhost:8080',
     'webpack/hot/only-dev-server',
-    './src/index.js'
+    './src/index.jsx'
   ],
 
   output: {
